### PR TITLE
161780567 user can make a comment

### PR DESCRIPTION
**Pull request by Jesse-efe**
#### Description
comments can be made on articles
comments can be replies to other comments
replies are threaded under their parent comment

#### Type of change
server
#### How Has This Been Tested?
tests were written and they all pass
#### Checklist:
N/A
####  Pivotal-tracker
161780567
#### Flow chart
comment on article

### CODE REVIEW

File: src/routes/api/articles/commentController.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    }
+  }
+
+  /**
+ *
+ *
+ * @static
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next
```
Comment (kayroy247) [inline src/routes/api/articles/commentController.js]: Nice job. Kindly replace the asterisks in the curly braces with the data types of the function parameters req, res and next.

Comment (Jesse-efe) [reply src/routes/api/articles/commentController.js]: ok thanks for pointing out

File: src/routes/api/articles/commentController.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ * @memberof comment
+ * @returns {void}
+ */
+  static async postComment(req, res, next) {
+    const { body } = req.body;
+    let articleId, parentCommentId;
+    if (req.params.articlesId === undefined) {
```
Comment (fuchodeveloper) [inline src/routes/api/articles/commentController.js]: The if statement should handle `undefined` without an extra check.

File: src/routes/api/articles/commentController.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      if (newComment) {
+        return res.status(201).json({
+          comment: {
+            id: newComment.id,
+            body: newComment.body
+          },
+        });
+      }
+    } catch (err) {
+      if (err.parent.code === '23503') {
```
Comment (fuchodeveloper) [inline src/routes/api/articles/commentController.js]: Please do a check for an alternative implementation approach not tied to Postgres.

**Review by fuchodeveloper — CHANGES_REQUESTED** [review]

File: src/routes/api/articles/commentController.js (hunk 3 above)
Comment (Jesse-efe) [reply src/routes/api/articles/commentController.js]: Ok

**Pull request merged**